### PR TITLE
Replaced Title Component by the one from V2 prototype.

### PR DESCRIPTION
**Pull request by rtxm**
This is part of the updates to #101.

### CODE REVIEW

File: kansha/board/boardconfig.py [hunk 1]
```diff
@@ -7,17 +7,19 @@
...
 #--
-
 from collections import namedtuple, OrderedDict
-from nagare import component, security, var
+
+from nagare import component
+from nagare import editor
+from nagare import i18n
+from nagare import security
+from nagare import validator, var
```
Comment (nagareproject) [inline kansha/board/boardconfig.py]: imports order

File: kansha/board/boardconfig.py [hunk 2]
```diff
@@ -7,17 +7,19 @@
...
 from nagare.i18n import _
 
 from kansha import notifications
 from kansha.menu import MenuEntry
-from ..label import comp as label
-from nagare import editor
-from nagare import validator
-from nagare import i18n
+from kansha.label import comp as label
```
Comment (nagareproject) [inline kansha/board/boardconfig.py]: Don't import the logic part (`comp`) of a component. The `__init__.py` of a component must be used to export its public API (a subset of the API in its `comp`). For example, in the `__init__.py` of `kansha.title`:
```python
from .comp import Title, EditableTitle

from . import view
```
and then, here:
```python
from kansha.title import Title
```
or
```python
from kasha import title
...
my_title = title.Title(...)
```

File: kansha/board/boardconfig.py [hunk 3]
```diff
@@ -63,9 +65,10 @@ def __init__(self, board):
         """
         self.board = board
         self.labels = []
-        for data in board.labels:
-            t = component.Component(label.LabelTitle(data))
-            l = component.Component(data, model='edit-color')
+        for lbl in board.labels:
+            t = component.Component(title.EditableTitle(lbl.get_title))
+            t.on_answer(lbl.set_title)
+            l = component.Component(lbl, model='edit-color')
```
Comment (nagareproject) [inline kansha/board/boardconfig.py]: Don't use variables of 1 or 2 characters or abbreviations

File: kansha/board/boardconfig.py [hunk 4]
```diff
@@ -63,9 +65,10 @@ def __init__(self, board):
         """
         self.board = board
         self.labels = []
-        for data in board.labels:
-            t = component.Component(label.LabelTitle(data))
-            l = component.Component(data, model='edit-color')
+        for lbl in board.labels:
+            t = component.Component(title.EditableTitle(lbl.get_title))
+            t.on_answer(lbl.set_title)
```
Comment (nagareproject) [inline kansha/board/boardconfig.py]: The `.on_awswer()` could be chain:
```python
t = component.Component(title.EditableTitle(lbl.get_title)).on_answer(lbl.set_title)
```

File: kansha/board/view.py [hunk 5]
```diff
@@ -121,7 +119,8 @@ def render_Board(self, h, comp, *args):
 
     with h.div(class_='board', style=style):
         with h.div(class_='header'):
-            h << self.title.render(h.AsyncRenderer())
+            with h.div(class_='board-title'):
+                h << self.title.render(h.AsyncRenderer(), 0 if security.has_permissions('edit', self) else 'readonly')
```
Comment (nagareproject) [inline kansha/board/view.py]: `0` -> `None`

File: kansha/board/view.py [hunk 6]
```diff
@@ -140,10 +139,10 @@ def render_Board_num_matches(self, h, comp, *args):
     if self.card_matches:
 
         if None in self.card_matches:
```
Comment (nagareproject) [inline kansha/board/view.py]: This condition is very strange. Why the above `if self.card_matches` condition is not enough?

File: kansha/card/comp.py [hunk 7]
```diff
@@ -80,7 +80,8 @@ def board(self):
     def reload(self, data=None):
         """Refresh the sub components
         """
-        self.title = component.Component(CardTitle(self))
+        self.title = component.Component(title.EditableTitle(self.get_title))
+        self.title.on_answer(self.set_title)
```
Comment (nagareproject) [inline kansha/card/comp.py]: idem

File: kansha/board/comp.py [hunk 8]
```diff
@@ -132,8 +131,8 @@ def many_user_render(h, number):
                       }
 
         # Title component
-        self.title = component.Component(BoardTitle(self))
-        self.title.on_answer(lambda v: self.title.call(model='edit'))
+        self.title = component.Component(title.EditableTitle(self.get_title))
+        self.title.on_answer(self.set_title)
```
Comment (nagareproject) [inline kansha/board/comp.py]: idem

File: kansha/card/view.py [hunk 9]
```diff
@@ -97,8 +99,9 @@ def render_card_edit(self, h, comp, *args):
 
     with h.div(class_='card-edit-form'):
         with h.div(class_='header'):
-            self.title.on_answer(lambda v: self.title.call(model='edit' if not self.title.model else None))
-            h << h.AsyncRenderer().div(self.title, component.Component(self.column, 'title'), class_="async-title")
+            with h.div(class_='title'):
+                h << self.title.render(h.AsyncRenderer(), 0 if security.has_permissions('edit', self) else 'readonly')
```
Comment (nagareproject) [inline kansha/card/view.py]: `0` -> `None`

File: kansha/checklist/comp.py [hunk 10]
```diff
@@ -25,23 +25,13 @@ def __init__(self):
...
-
 class ChecklistItem(object):
 
     def __init__(self, id_, data=None):
         self.id = id_
         data = data if data is not None else self.data
-        self.title = component.Component(ChecklistItemTitle(self))
-        self.title.on_answer(lambda v: self.title.call(model='edit' if not self.title.model else None))
+        self.title = component.Component(title.EditableTitle(self.get_title, placeholder=i18n._(u'Enter task')))
+        self.title.on_answer(self.set_title)
```
Comment (nagareproject) [inline kansha/checklist/comp.py]: Idem

File: kansha/checklist/comp.py [hunk 11]
```diff
@@ -75,20 +65,12 @@ def __init__(self, id_, data=None):
         data = data if data is not None else self.data
         self.items = [component.Component(ChecklistItem(item.id, item)) for item in data.items]
 
-        self.title = component.Component(ChecklistTitle(self))
-        self.title.on_answer(self.handle_answer)
+        self.title = component.Component(title.EditableTitle(self.get_title, placeholder=i18n._(u'Enter title')))
```
Comment (nagareproject) [inline kansha/checklist/comp.py]: Idem

File: kansha/column/comp.py [hunk 12]
```diff
@@ -40,8 +40,8 @@ def __init__(self, id_, board, card_extensions, search_engine, services_service,
         self.search_engine = search_engine
         self.card_extensions = card_extensions
         self.body = component.Component(self, 'body')
-        self.title = component.Component(ColumnTitle(self))
-        self.title.on_answer(lambda v: self.title.call(model='edit'))
+        self.title = component.Component(title.EditableTitle(self.get_title))
```
Comment (nagareproject) [inline kansha/column/comp.py]: Idem

File: kansha/column/view.py [hunk 13]
```diff
@@ -96,16 +96,16 @@ def render_column_overlay(self, h, comp, *args):
 
 @presentation.render_for(Column, model='header')
 def render_column_header(self, h, comp, *args):
-    if self.card_counter.model != 'edit':
-        h << self.title
-    if self.title.model != 'edit':
-        h << self.card_counter
+    with h.div(class_='title'):
+        h << self.title.render(h.AsyncRenderer(), 0 if security.has_permissions('edit', self) else 'readonly')
```
Comment (nagareproject) [inline kansha/column/view.py]: `0` -> `None`

File: kansha/comment/comp.py [hunk 14]
```diff
@@ -132,7 +132,7 @@ def add(self, v):
                                   author=user.data, creation_date=datetime.datetime.utcnow())
             session.add(comment)
             session.flush()
-            data = {'comment': v.strip(), 'card': self.parent.title().text}
+            data = {'comment': v.strip(), 'card': self.parent.get_title()}
```
Comment (nagareproject) [inline kansha/comment/comp.py]: `self.parent` -> `self.card`

File: static/css/themes/kansha_flat/board.css [hunk 15]
```diff
@@ -12,18 +12,18 @@
...
+.board .board-title {
     color: #737373;
     font: normal 33px OpenSansLight;
 }
 
-.board .boardTitle a,
-.board .boardTitle a:hover {
+.board .board-title a,
+.board .board-title a:hover {
     color: #737373;
```
Comment (nagareproject) [inline static/css/themes/kansha_flat/board.css]: I think  `#737373` must be the default text color, set on `body` and inherit everywhere

File: kansha/board/view.py (hunk 6 above)
Comment (rtxm) [inline kansha/board/view.py]: `card_matches`  is a list which contains the ids of the cards that match the filter, or only None if no cards match the filter. If `card_matches` itself evaluates to False, that means the filter is off.

File: kansha/board/boardconfig.py (hunk 4 above)
Comment (rtxm) [inline kansha/board/boardconfig.py]: Don't we apply any limit on line width?

File: kansha/card/comp.py (hunk 7 above)
Comment (rtxm) [inline kansha/card/comp.py]: line too long?

File: kansha/board/boardconfig.py (hunk 4 above)
Comment (Net-ng) [inline kansha/board/boardconfig.py]: > Le 3 déc. 2015 à 14:00, Romuald Texier-Marcadé <notifications@github.com> a écrit :
> 
> In kansha/board/boardconfig.py:
> 
> > @@ -63,9 +65,10 @@ def __init__(self, board):
> >          """
> >          self.board = board
> >          self.labels = []
> > -        for data in board.labels:
> > -            t = component.Component(label.LabelTitle(data))
> > -            l = component.Component(data, model='edit-color')
> > +        for lbl in board.labels:
> > +            t = component.Component(title.EditableTitle(lbl.get_title))
> > +            t.on_answer(lbl.set_title)
> 
> Don't we apply any limit on line width?

We do. But not 72 like in the « old ages » of the first version of the PEP8.

We follow the latest version, with a max length of 99:

"
Some teams strongly prefer a longer line length. For code maintained exclusively
or primarily by a team that can reach agreement on this issue, it is okay to
increase the nominal line length from 80 to 100 characters (effectively increasing
the maximum length to 99 characters), provided that comments and docstrings are
still wrapped at 72 characters.
"

(https://www.python.org/dev/peps/pep-0008/#maximum-line-length)

File: kansha/board/view.py (hunk 6 above)
Comment (nagareproject) [inline kansha/board/view.py]: > Le 3 déc. 2015 à 13:57, Romuald Texier-Marcadé <notifications@github.com> a écrit :
> 
> In kansha/board/view.py:
> 
> > @@ -140,10 +139,10 @@ def render_Board_num_matches(self, h, comp, *args):
> >      if self.card_matches:
> >  
> >          if None in self.card_matches:
> 
> card_matches is a list which contains the ids of the cards that match the filter, or only None if no cards match the filter. If card_matches itself evaluates to False, that means the filter is off.

Mixing several semantics is not correct

Comment (rtxm) [inline kansha/board/view.py]: See #108.

**Comment by rtxm** [comment]
All comments addressed, @nagareproject.

**Pull request merged**